ui/src/reducers/index.js

```diff
@@ -99,19 +104,43 @@ export default (state = initialState, action) => {
           data: action.result
         }
       };
-    case APP_DETAILS_SUCCESS:
-      const fetchedAppDetails = [];
-      action.result.forEach((appDetail) => {
-        fetchedAppDetails.push(appDetail);
+    case APPS_FAILURE:
+      return {
+        ...state,
+        isAppsRequestFailed: true
+      };
+    case APP_REQUEST:
+      return {
+        ...state
+      };
+    case APP_SUCCESS:
```

Comment:
I think you included this because we do the translation for the apps request in the reducer also. But because there is a React Component (the checkbox) required in the appversions table @damienomurchu did the translation in the component rather than in the reducer. We discussed moving the apps rows translation to the component again, but it's not necessary that we do that. For this case, as we're already doing it in the component we don't need to do it here.

Reply:
In that case we probably don't even need to store the rows value in the state as we're creating that in the component and can use the json data to build it there.

Reply:
Ah, I originally had it like that but changed it to keep the component changes minimal. I kept the component code mostly the same:

https://github.com/aerogear/mobile-security-service/pull/95/files#diff-2aac7a8dcccd74e488dbea5673dfc9c2R14

I will change it back.

Reply:
I think we might need to store the row values in the state for sorting.

Reply:
@craicoverflow yep, you're 100% right. Resolving this one and there's no action required.

Reply:
> I think we might need to store the row values in the state for sorting.

I think you're right. We only get the re-rendering of the table component with redux when the state mapped in `mapPropsToState` in the component changes. 

With the `AppVersionTableContainer` we need to modify the row data to create the checkbox, and we shouldn't be storing React components (like Checkbox'es) in the store, so I think we need logic in both places; in the reducer to store the row data in the state, and in the component to re-render the row data when it changes (& generate the checkboxes)